RockLib.Messaging.Kafka/KafkaReceiver.cs

```diff
@@ -117,11 +124,14 @@ public KafkaReceiver(string name, string topic, ConsumerConfig consumerConfig, b
             builder.SetStatisticsHandler(OnStatisticsEmitted);
 
             _consumer = new Lazy<IConsumer<string, byte[]>>(() => builder.Build());
-            _trackingCollection = new BlockingCollection<KafkaReceiverMessage>();
+            _trackingCollection = CreateMessageTrackingCollection(messageBufferSize);
 
             _schemaIdRequired = schemaIdRequired;
         }
 
+        private static BlockingCollection<KafkaReceiverMessage> CreateMessageTrackingCollection(int size) =>
+            new(size <= 0 ? int.MaxValue : size);
```

Comment:
net485 friendly?

Reply:
Yes - https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.blockingcollection-1.-ctor?view=netframework-4.8#system-collections-concurrent-blockingcollection-1-ctor(system-int32)

Reply:
I was looking at the implementation of [`BlockingCollection`](https://source.dot.net/#System.Collections.Concurrent/System/Collections/Concurrent/BlockingCollection.cs), and the `boundedCapacity` value being set to `int.MaxValue`....it doesn't look like this is doing some kind of pre-allocation of an internal array (like what happens when you pass in an initial capacity to a `List<>`), but....it also seems a bit weird to pass in a maximum value. It also seems like this value isn't being used internally by the collection; it's only used when someone reads the `BoundedCapacity` value. So I'm just looking for some insight into why this was changed to this approach.

Reply:
When a size <= 0 is provided, the intent was to replicate an unbounded collection (i.e. the default constructor behavior). The alternative was to have the ternary choose between the default constructor and the bounded one. Easy change if that approach is felt as the better one.

Reply:
I really don't have an opinion on this. It just stuck out to me because it made me wonder, are we really pre-allocation that many items when we create it? Which, when I looked at the code, it doesn't appear like that happens. But, just setting it to MaxSize, by the docs it says that we'll basically never block, but....there is a limit to the amount of processors on a machine, so would we start to tax a system if we do go over a certain amount?